src/device/usbd.c

```diff
@@ -484,39 +530,53 @@ static bool process_control_request(uint8_t rhport, tusb_control_request_t const
 
     //------------- Endpoint Request -------------//
     case TUSB_REQ_RCPT_ENDPOINT:
-      // Non standard request is not supported
-      TU_VERIFY( TUSB_REQ_TYPE_STANDARD == p_request->bmRequestType_bit.type );
+    {
+      bool drvHandled = false;
+      if(drvid != INVALID_DRV_TOKEN) {
+        usbd_control_set_complete_callback(usbd_class_drivers[drvid].control_complete);
 
-      switch ( p_request->bRequest )
-      {
-        case TUSB_REQ_GET_STATUS:
-        {
-          uint16_t status = usbd_edpt_stalled(rhport, tu_u16_low(p_request->wIndex)) ? 0x0001 : 0x0000;
-          tud_control_xfer(rhport, p_request, &status, 2);
-        }
-        break;
+        drvHandled = usbd_class_drivers[drvid].control_request(rhport, p_request);
+      }
 
-        case TUSB_REQ_CLEAR_FEATURE:
-          if ( TUSB_REQ_FEATURE_EDPT_HALT == p_request->wValue )
+      if( TUSB_REQ_TYPE_STANDARD == p_request->bmRequestType_bit.type ) {

```

Comment:
Adds condition that the "standard" requests are only handled when the type is standard (after removing the TU_VERIFY of this fact.